tests/eos-test-utils.c

```diff
@@ -0,0 +1,1960 @@
+               GFile *hw_file,
+               GFile *avahi_emulator_definitions_dir,
+               GFile *quit_file,
+               const gchar *osname,
+               CmdAsyncStuff *cmd,
+               GError **error)
+{
+  SSDEF;
+  g_auto(GStrv) argv = NULL;
+  g_auto(GStrv) envp = NULL;
+  g_autofree gchar *bash_script_path = NULL;
+  g_autoptr(GMainLoop) loop = g_main_loop_new (NULL, FALSE);
+  WatchUpdater wu = { loop, 0u };
+  guint id = g_bus_watch_name (G_BUS_TYPE_SESSION,
+                               "com.endlessm.Updater",
+                               G_BUS_NAME_WATCHER_FLAGS_NONE,
+                               com_endlessm_updater_appeared,
+                               NULL,
+                               &wu,
+                               NULL);
+
+  wu.id = id;
+  argv = generate_strv (EOS_UPDATER_BINARY,
+                        NULL);
+  envp = generate_strv (SS (envvar ("EOS_UPDATER_TEST_UPDATER_CONFIG_FILE_PATH",
+                                    SS (g_file_get_path (config_file)))),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATER_AVAHI_SERVICES_DIR",
+                                    SS (g_file_get_path (avahi_services_dir)))),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATER_CUSTOM_DESCRIPTORS_PATH",
+                                    SS (g_file_get_path (hw_file)))),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATER_AVAHI_EMULATOR_DEFINITIONS_DIR",
+                                    SS (g_file_get_path (avahi_emulator_definitions_dir)))),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATER_DEPLOYMENT_FALLBACK",
+                                    "yes")),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATER_QUIT_FILE",
+                                    SS (g_file_get_path (quit_file)))),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATER_USE_SESSION_BUS",
+                                    "yes")),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATER_USE_AVAHI_EMULATOR",
+                                    "yes")),
+                        SS (envvar ("EOS_UPDATER_TEST_UPDATER_OSTREE_OSNAME",
+                                    osname)),
+                        SS (envvar ("OSTREE_SYSROOT",
+                                    SS (g_file_get_path (sysroot)))),
+                        SS (envvar ("OSTREE_REPO",
+                                    SS (g_file_get_path (repo)))),
+                        NULL);
+
+  bash_script_path = g_strdup (g_getenv ("EOS_CHECK_UPDATER_GDB_BASH_PATH"));
+  if (bash_script_path != NULL)
+    {
+      g_autoptr(GFile) path = g_file_new_for_path (bash_script_path);
+
+      if (!generate_bash_script (path, argv, envp, error))
+        return FALSE;
+
+      g_printerr ("Bash script %s generated, chmod a+x it an run it, make check will continue when com.endlessm.Updater appears on the test session bus\n",
+                  bash_script_path);
```

Comment:
This is only executed if you run checks like `make check EOS_UPDATER_GDB=$PWD/eos-gdb.sh`. So it makes it easy to debug `eos-updater`. Otherwise we just execute `eos-updater` directly, without `gdb`